### PR TITLE
Fix board edit/update validations

### PR DESCRIPTION
**Pull request by mfrister**
https://www.openproject.org/issues/2615

### CODE REVIEW

File: spec/controllers/boards_controller_spec.rb [hunk 1]
```diff
@@ -114,4 +114,53 @@
...
 
   end
+
+  describe :update do
+    let!(:board) { FactoryGirl.create(:board) }
+
+    before do
+      @controller.should_receive(:authorize)
+      # @controller.should_receive(:find_project_by_project_id) do
+      #   @controller.instance_variable_set(:@project, project)
```
Comment (ulferts) [inline spec/controllers/boards_controller_spec.rb]: those commented out lines should be removed

File: spec/controllers/boards_controller_spec.rb [hunk 2]
```diff
@@ -114,4 +114,53 @@
...
+      # end
+    end
+
+    describe 'w/ the params beeing valid' do
+
+      before do
+        Board.any_instance.should_receive(:update_attributes).and_return(true)
+
+        as_logged_in_user user do
+          post :update, id: board.id, project_id: board.project_id, board: 'fake param'
```
Comment (ulferts) [inline spec/controllers/boards_controller_spec.rb]: Why do you provide valid parameters when you have that 

<pre>
Board.any_instance.should_receive(:update_attributes).and_return(true)
</pre>


statement. What I am getting at is that the mentioned statement should be changed. This test does not assure that the controller updates the correct board.

File: spec/controllers/boards_controller_spec.rb [hunk 3]
```diff
@@ -79,7 +79,7 @@
         end
       end
 
-      it 'should redirect to the settings page if successful' do
+      it 'should render the new template' do
```
Comment (ulferts) [inline spec/controllers/boards_controller_spec.rb:82]: Thx for correcting this.

**Comment by ulferts** [comment]
Functionality looks good. Fix the tests and I will merge.

**Comment by mfrister** [comment]
Thanks for the comments!

Didn't find an easy way to stub a specific instance of the board, so I just provide valid params and later check them. Removed the Board.any_instance stubbing and the commented out lines.

File: spec/controllers/boards_controller_spec.rb (hunk 1 above)
Comment (mfrister) [inline spec/controllers/boards_controller_spec.rb]: Uh.

**Pull request merged**